### DISCUSSION TITLE
Similar methods, different parameters

### DISCUSSION THREAD

**Discussion by lamazze** [sebastienros/jint#1540] [Q&A] [answered]
Hello,

I've shared a ExcelWorksheet (EEPlus) object by calling "engine.SetValue()".
Everything works fine except the Cell property returns an ExcelRange class that has two identical methods (not sure if my vocabulary is right):
- public ExcelRange this[string Address]
- public ExcelRange this[int Row, int Col]

If in Jint, I use "sheet.Cells["A1"].Value, I have the content of the cell.
But if I use "sheet.Cells[1, 1].Value" which should return the same thing, I have the following error message: "1 is undefined"

I'm not sure but I think it has to do with the fact it expects a string and stop there ?

Thank you
lamazze

**Comment by lahma** [comment, marked as answer]
Two-dimensional indexers probably just aren't supported, Jint understands the basic indexer.

**Comment by lamazze** [comment]
Thank you lahma. Do you reckon it's worth doing a feature request ?

**Reply by lahma** [reply]
I'd suggest creating a PR instead if you need the feature. Pull request with test case(s) and required code tweaks is the best option to get the feature landed 👍🏻

**Reply by SebastianStehle** [reply]
How would it look in javascript?

**Reply by lahma** [reply]
Hah, good catch. Maybe my morning brain. Might be easiest to have an extension method to handle this. Like Cells(1, 1)?

**Reply by lahma** [reply]
A JS equivalent would probably mean supplying an array to indexer call, like `sheet.Cells[ ([1, 1]) ]` and Jint would need to interpret that.

**Comment by lamazze** [comment]
For your information, with the stable version I can use: sheet.Cells["A1"].Value and not sheet.Cells[0, 0].Value.
But with the beta, nothing is working:
- When I use sheet.Cells["A1"].Value, the error is "Array was not a one-dimensional array"
- When I use sheet.Cells[0, 0].Value, the error is "Address is not valid"

**Reply by lahma** [reply]
In this case it should be technically possible to disambiguate the string indexer, unless it has more overloads that could cause more ambiguity.

**Reply by viruscamp** [reply]
Javascript grammar won't do the thing you want. Maybe it's [comma operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator) .
```js
sheet.Cells[1, 3].Value;
//is just
sheet.Cells[3].Value;
```

**Reply by viruscamp** [reply]
It's possible to separate `sheet.Cells["idx"]` and `sheet.Cells[1]`.

And can we export the underlying methods ?
```js
sheet.get_Item(1,3).Value;
sheet.set_Item(3,4, sth);
```

**Reply by lamazze** [reply]
@viruscamp using get_Item "solves" the problem.

sheet.Cells["A1"].Value returns the content of the cell
sheet.Cells[1, 1].Value returns "1 is undefined"
sheet.Cells[1][1].Value returns "1 is undefined"
sheet.Cells.get_Item(1, 1).Value returns the content of the cell
sheet.Cells.get_Item("A1").Value returns the content of the cell

**Reply by lamazze** [reply]
I forgot to mention that these are the result with v2 of JINT.
With 3 beta, nothing works:
sheet.Cells["A1"].Value returns "Array was not a one-dimensional array"
sheet.Cells[1, 1].Value returns "Address is not valid"
sheet.Cells[1][1].Value returns "Address is not valid"
sheet.Cells.get_Item(1, 1).Value returns "Array was not a one-dimensional array"
sheet.Cells.get_Item("A1").Value returns "Array was not a one-dimensional array"

**Reply by lahma** [reply]
OK so I think the expected outcomes for 3.x would be:

```c#
sheet.Cells["A1"].Value returns "Array was not a one-dimensional array" // SHOULD WORK
sheet.Cells[1, 1].Value returns "Address is not valid" // SHOULD NOT WORK AS NOT JS COMPATIBLE
sheet.Cells[1][1].Value returns "Address is not valid" // SHOULD WORK
sheet.Cells.get_Item(1, 1).Value returns "Array was not a one-dimensional array" // SHOULD WORK
sheet.Cells.get_Item("A1").Value returns "Array was not a one-dimensional array" // SHOULD WORK
```

**Comment by viruscamp** [comment]
```javascript
sheet.Cells[1][1].Value returns "Address is not valid" // I donot like this
sheet.Cells.get_Item(1, 1).Value returns "Array was not a one-dimensional array" // I think it's enough
```
It will import a curring function like object `sheet.Cells[1]`, and will make more harder overloads.

**Reply by viruscamp** [reply]
Assume we have such c# class:
```c#
class TestIndexer
{
  public static double this[int idx] { get; set; }
  public static string this[int idx, string ref] { get; set; }
}
```
In js, the grammar `sheet.Cells[1][1].Value returns "Address is not valid"` will make  ambiguity, 
`TestIndexer[0]` is a double or `TestIndexer[0]['abc']` is a string?
